src/template.c

```diff
@@ -46,7 +62,9 @@ std::vector<std::string> template_dfa(CharacterVector x) {
       switch (c) {
       case '{':
         state = code;
-        pieces.push_back(input.substr(pieceStartIdx, i - pieceStartIdx - 1));
+        str = PROTECT(Rf_mkCharLenCE(input + pieceStartIdx, i - pieceStartIdx - 1, CE_UTF8));
+        REPROTECT(pieces = str_sxp_set(pieces, pieces_num++, str), pieces_idx);
```

Comment:
This `REPROTECT` code looks a bit weird, but it needed to be written this way to that rchk does not complain, we use very similar code in [glue](https://github.com/tidyverse/glue/blob/29eb0ff076810b5fbfac378072bff3ab77eba25b/src/glue.c#L149-L150)

Reply:
Just curious - does the `UNPROTECT` affect this `REPROTECT`, or the previous `PROTECT`?

Reply:
Oh wait, I think I get it now.

Reply:
Can you add a comment with a quick explanation about this?